constructor/main.py

```diff
@@ -76,6 +76,67 @@ def get_output_filename(info):
         ext,
     )
 
+# Validate frozen environments
+def validate_frozen_envs(info, exe_type, exe_version):
+    """Validate frozen environments.
+
+        Checks:
+        - No conflicts between freeze_base/freeze_env and extra_files for same environment
+        - conda-standalone 25.5.x is not used (has known issues)
+        - Warns if conda-standalone < 25.5.0 (frozen files will be ignored)
+
+        Stores frozen environment info in `_frozen_markers` dict.
+    """
+    def get_frozen_env_from_path(dest: str) -> str | None:
+        """Extract environment name from frozen marker destination path."""
+        parts = Path(dest).parts
+        if parts == ("conda-meta", "frozen"):
+            return "base"
+        if len(parts) == 4 and parts[0] == "envs" and parts[-2:] == ("conda-meta", "frozen"):
+            return parts[1]
+        return None
+
+    # Collect environments using freeze_base/freeze_env
+    frozen_envs = {}
+    if info.get("freeze_base"):
+        frozen_envs["base"] = {
+            "method": "freeze_base",
+            "config": info["freeze_base"],
+        }
+    for env_name, env_config in info.get("extra_envs", {}).items():
+        if env_config.get("freeze_env"):
+            frozen_envs[env_name] = {
+                "method": "freeze_env",
+                "config": env_config["freeze_env"],
+            }
+
+    # Check for conflicts with extra_files
+    for file in info.get("extra_files", []):
+        if isinstance(file, dict):
+            for dest in file.values():
+                env = get_frozen_env_from_path(dest)
+                if env and env in frozen_envs:
+                    raise RuntimeError(
+                        f"Environment '{env}' has frozen markers from both "
+                        f"'{'freeze_base' if env == 'base' else 'freeze_env'}' and 'extra_files'. "
+                        "Please use only one method to provide frozen markers for each environment.")
+
+    info["_frozen_markers"] = frozen_envs
```

Comment:
What do we need this for? All information we need for frozen environments is already stored in `info`.